executor/projection.go

```diff
@@ -290,7 +296,9 @@ func (e *ProjectionExec) drainOutputCh(ch chan *projectionOutput) {
 
 // Close implements the Executor Close interface.
 func (e *ProjectionExec) Close() error {
-	if e.isUnparallelExec() {
+	// if e.baseExecutor.Open returns error, e.childResult will be nil, see https://github.com/pingcap/tidb/issues/24210
+	// for more information
+	if e.isUnparallelExec() && e.childResult != nil {
```

Comment:
We should also check every usage of `childResult`, and wrap them with `if e.childResult != nil`.

Reply:
I don't think we should check at every usage, because if open returned error, only Close will be called.

Reply:
remove this if below [L318](https://github.com/pingcap/tidb/pull/24231/files#diff-ff427415020e8206064bbb4bba684d29e8aa5e8bba1b990164c5460908dab52dR318)